web/packages/teleport/src/services/api/api.ts

```diff
@@ -316,6 +316,7 @@ export function getNoCacheHeaders() {
   };
 }
 
+// TODO(kimlisa): DELETE IN v19.0 (csrf)
```

Comment:
I don't think we can remove this yet, as we still need it for SSO flows. Right?

Reply:
when i looked up the usage of this, the remaining use of this was for plugin related stuff.

for SSO, we extract the csrf from the cookie

Reply:
There is no session cookie until you log in and SSO completes.   

I could be missing something, but I believe we're using this as a form of pre-auth token and storing it in the backend in the pending SSO login, which ensures that the browser that completes the SSO flow is the same as the browser that started it.

Reply:
i removed the delete comment just to be safe.

> There is no session cookie until you log in and SSO completes.

it looks like we do set a [CSRF cookie](https://github.com/gravitational/teleport/blob/56153e94a6fc388c5f84a6d98102ed30b0272a02/lib/web/apiserver.go#L675) on serving the index page (before login)